src/main/java/ratspeak/data/TaskList.java

```diff
@@ -0,0 +1,148 @@
+package ratspeak.data;
+
+import ratspeak.exception.DukeException;
+import ratspeak.storage.Storage;
+import ratspeak.task.Task;
+
+import java.util.ArrayList;
+
+/**
+ * represents a task list that stores  a list of tasks
+ * performs add, delete, mark, unmark
+ */
+public class TaskList {
+    private ArrayList<Task> tasklist;

```

Comment:
Consider changing the variable name tasklist to taskList to follow Java naming conventions.